### PR TITLE
feat: add auction countdown modal

### PR DESCRIPTION
**Pull request by ThomasRalee**
## This PR:
- resolves [FE-324](https://linear.app/injective/issue/FE-324/watch-the-burn-popup)

## Screenshot:
image

### CODE REVIEW

**Comment by albertchon** [comment]
Instead of 00 days, would change to 0 days. Also can you send a screenshot of the entire page so I can see what this looks like?

**Comment by cmhchoi** [comment]
https://user-images.githubusercontent.com/9430957/146151013-1f65a495-659b-4fb9-bead-0bdcf267a704.mp4

File: components/partials/modals/auction-countdown.vue [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    setIntervalForNow() {
+      setInterval(() => {
+        this.now = Date.now()
+      }, 1000)
+    },
+
+    closeModal() {
+      const { auctionModuleState } = this
+
+      const round = auctionModuleState?.auctionRound || hardcodedAuctionRound
```
Comment (bangjelkoski) [inline components/partials/modals/auction-countdown.vue:130]: If the round is 0 (or `auctionModuleState` is undefined) don't show anything, don't use hardcoded values.

File: layouts/default.vue [hunk 2]
```diff
@@ -78,6 +84,16 @@ export default Vue.extend({
         //
       })
       .catch(this.$onRejected)
+
+    if (
+      SHOW_AUCTION_COUNTDOWN &&
+      !this.$accessor.auction.auctionsViewed.includes(hardcodedAuctionRound) &&
+      hardcodedEndTime > Date.now()
```
Comment (bangjelkoski) [inline layouts/default.vue:91]: Change to include `auction_ending_timestamp - 24 hours` as `hardcodedEndTime`.

File: store/auction.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import { actionTree } from 'typed-vuex'
+import { AuctionModuleState } from '@injectivelabs/chain-consumer'
+import { fetchAuctionModuleState } from '~/app/services/auction'
+
+// hard code auction round to optimize performance by skipping fetchAuction api call if auctionsViewed contains auctionRound
+export const hardcodedAuctionRound = 1
+export const hardcodedEndTime = 1639573200 * 1000
```
Comment (bangjelkoski) [inline store/auction.ts:7]: Remove these, no need to skip `fetchAuction`. We can skip calling this function if the flag `SHOW_AUCTION_COUNTDOWN` is set to true

**Review by bangjelkoski — APPROVED** [review]

**Pull request merged**